src/main/java/io/vertx/core/http/Http2Settings.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright (c) 2011-2013 The original author or authors
+ *  ------------------------------------------------------
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.core.http;
+
+import io.netty.handler.codec.http2.Http2CodecUtil;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.annotations.GenIgnore;
+import io.vertx.core.impl.Arguments;
+import io.vertx.core.json.JsonObject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * HTTP2 settings, the settings is initialized with the default HTTP/2 values.<p>
+ *
+ * The settings expose the parameters defined by the HTTP/2 specification, as well as extra settings for
+ * protocol extensions.
+ *
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+@DataObject(generateConverter = true)
+public class Http2Settings {
+
+  public static final long DEFAULT_HEADER_TABLE_SIZE = 4096;
```

Comment:
good catch.

most of them are the default spec values.

so I think that’s what we should mention indeed.


> On Apr 6, 2016, at 10:19 AM, Clement Escoffier <notifications@github.com> wrote:
> 
> In src/main/java/io/vertx/core/http/Http2Settings.java <https://github.com/eclipse/vert.x/pull/1353#discussion_r58667503>:
> 
> > +
> > +import java.util.HashMap;
> > +import java.util.Map;
> > +
> > +/**
> > + * HTTP2 settings, the settings is initialized with the default HTTP/2 values.<p>
> > + *
> > + * The settings expose the parameters defined by the HTTP/2 specification, as well as extra settings for
> > + * protocol extensions.
> > + *
> > + * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
> > + */
> > +@DataObject(generateConverter = true)
> > +public class Http2Settings {
> > +
> > +  public static final long DEFAULT_HEADER_TABLE_SIZE = 4096;
> Should we have javadoc on these public constants ?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/eclipse/vert.x/pull/1353/files/c7a6a1ce8a1da75fc0edcccf686f25b82162b149#r58667503>